tests/integration-tests/tests/api_infrastructure/test_api_infrastructure.py

```diff
@@ -0,0 +1,131 @@
+        return file_content
+
+    yield _create_api_infrastructure
+    factory.delete_all_stacks()
+
+
+@pytest.mark.skip_regions(["cn-north-1", "cn-northwest-1"])  # TODO No Lambda container support in China regions
+@pytest.mark.usefixtures("os", "instance", "scheduler")
+def test_api_infrastructure_with_default_parameters(
+    region, api_definition_s3_uri, public_ecr_image_uri, api_infrastructure_stack_factory
+):
+    """Test that creating the API Infrastructure stack with the defaults correctly sets up the Lambda and APIGateway API
+
+    :param region: the region where the stack is run
+    :api_definition_s3_uri: a fixture, the S3 URI of the API definition
+    :public_ecr_image_uri: a fixture, the URI of the published Lambda Docker image
+    :api_infrastructure_stack_factory: a fixture, the factory used to create the regional API Infrastructure stack
+    """
+    parameters = [
+        {"ParameterKey": "ApiDefinitionS3Uri", "ParameterValue": api_definition_s3_uri},
+        {"ParameterKey": "PublicEcrImageUri", "ParameterValue": public_ecr_image_uri},
+    ]
+    stack = api_infrastructure_stack_factory(region, parameters)
+
+    lambda_client = boto3.client("lambda", region_name=region)
+    apigateway_client = boto3.client("apigateway", region_name=region)
+
+    parallelcluster_lambda_name = stack.cfn_resources["ParallelClusterFunction"]
+    parallelcluster_lambda_arn = stack.cfn_outputs["ParallelClusterLambdaArn"]
+    parallelcluster_api_copied_image_uri = stack.cfn_outputs["UriOfCopyOfPublicEcrImage"]
+
+    parallelcluster_api_id = stack.cfn_resources["ApiGatewayApiWithoutCustomDomain"]
+    parallelcluster_api_url = stack.cfn_outputs["ParallelClusterApiInvokeUrl"]
+
+    with soft_assertions():
+        _assert_parallelcluster_lambda(
+            client=lambda_client,
+            lambda_name=parallelcluster_lambda_name,
+            lambda_arn=parallelcluster_lambda_arn,
+            lambda_image_uri=parallelcluster_api_copied_image_uri,
+        )
+
+        _assert_parallelcluster_api(
+            client=apigateway_client, api_id=parallelcluster_api_id, api_url=parallelcluster_api_url
+        )
+
+
+def _assert_parallelcluster_lambda(client, lambda_name, lambda_arn, lambda_image_uri):
+    """Check that the ParallelCluster Lambda is correctly configured
+
+    :param client: the Lambda client
+    :param lambda_name: the name of the ParallelCluster Lambda
+    :param lambda_arn: the ARN of the ParallelCluster Lambda
+    :param lambda_image_uri: the URI of the local copy of the ParallelCluster Lambda Docker image
+    """
+    lambda_resource = client.get_function(FunctionName=lambda_name)
+
+    assert_that(lambda_resource["Configuration"]["FunctionArn"]).is_equal_to(lambda_arn)
+    assert_that(lambda_resource["Tags"]).contains("parallelcluster:version")
+    assert_that(lambda_resource["Code"]["ImageUri"]).is_equal_to(lambda_image_uri)
+
+
+def _assert_parallelcluster_api(client, api_id, api_url):
```

Comment:
can we also assert authorizer config and tracing?
I'm also wondering how hard would it be to perform a simple end-to-end request to the simplest api operation such as ListClusters and just assert the code is 200.

Reply:
1. I have tried the library suggested in [this GitHub issue](https://github.com/boto/boto3/issues/1246) unsuccessfully. I have implemented the call with Botocore/Requests
1. Regarding the authorizer config, I am unsure about what to assert. Specifically, I tried [get_deployment](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/apigateway.html#APIGateway.Client.get_deployment), and [get_resource](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/apigateway.html#APIGateway.Client.get_resource)/[get_resources](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/apigateway.html#APIGateway.Client.get_resources) methods and I don't see an `authorizationType` in the response. Looking at the processed template, I see the following in the [body of the REST API resource](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-restapi.html#cfn-apigateway-restapi-body), which is the OpenAPI specification defining the RESTful API:

```
"security": [
    {
        "AWS_IAM": []
    }
]
```